src/components/EventBlocks/index.tsx

```diff
@@ -0,0 +1,268 @@
+            deviceHasHover={deviceHasHover}
+            // Only the first day for a meeting can be tab focused
+            canBeTabFocused={canBeTabFocused && i === 0}
+          />
+        );
+      })}
+    </div>
+  );
+}
+
+type MeetingDayBlockProps = {
+  color: string | undefined;
+  day: string;
+  name: string;
+  period: Period;
+  days: string[];
+  sizeInfo: EventBlockPosition;
+  canBeTabFocused: boolean;
+  includeTime: boolean;
+  includeDetailsPopover: boolean;
+  includeContent: boolean;
+  isSelected: boolean;
+  onSelect: (newIsSelected: boolean) => void;
+  deviceHasHover: boolean;
+};
+
+function MeetingDayBlock({
+  color,
+  day,
+  name,
+  period,
+  days,
+  sizeInfo,
+  canBeTabFocused = false,
+  includeTime,
+  includeDetailsPopover,
+  includeContent,
+  isSelected,
+  onSelect,
+  deviceHasHover,
+}: MeetingDayBlockProps): React.ReactElement {
+  const tooltipId = useId();
+  const contentClassName = getContentClassName(color);
+  const outerRef = React.useRef<HTMLDivElement>(null);
+  const handleRootClose = (): void => {
+    if (isSelected) onSelect(false);
+  };
+  useRootClose(outerRef, handleRootClose, {
+    disabled: !isSelected,
+  });
+
+  const [isHovered, setIsHovered] = React.useState(false);
+  const BlockElement = canBeTabFocused ? 'button' : 'div';
+  return (
+    <div ref={outerRef}>
+      <BlockElement
+        className={classes(
+          'meeting',
+          contentClassName,
+          'default',
+          day,
+          isSelected && 'meeting--selected'
+        )}
+        style={{
+          top: `${((period.start - OPEN) / (CLOSE - OPEN)) * 100}%`,
+          height: `${
+            (Math.max(15, period.end - period.start) / (CLOSE - OPEN)) * 100
+          }%`,
+          width: `${20 / sizeInfo.rowSize}%`,
+          left: `${
+            DAYS.indexOf(day) * 20 + sizeInfo.rowIndex * (20 / sizeInfo.rowSize)
+          }%`,
+          ...({
+            '--meeting-color': color,
+          } as React.CSSProperties),
+        }}
+        id={tooltipId}
+        onClick={(e: React.MouseEvent): void => {
+          e.stopPropagation();
+          onSelect(true);
+        }}
+        onFocus={
+          canBeTabFocused
+            ? (e: React.FocusEvent): void => {
+                e.stopPropagation();
+                setIsHovered(true);
+              }
+            : undefined
+        }
+        onBlur={
+          canBeTabFocused
+            ? (e: React.FocusEvent): void => {
+                e.stopPropagation();
+                setIsHovered(false);
+              }
+            : undefined
+        }
+        tabIndex={canBeTabFocused ? 0 : -1}
+      >
+        {includeContent && (
+          <div className="meeting-wrapper">
+            <div className="ids">
+              <span className="event-id">{name}</span>
+            </div>
+            {includeTime && (
+              <span className="period">{periodToString(period)}</span>
+            )}
+          </div>
```

Comment:
There's a good amount of overlap between `TimeBlocks` and `EventBlocks` that I think we can make a parent component out of them. In fact, pretty much only this part of the code and a few more are different between the two.

Let's convert `TimeBlocks` into a parent component with all the overlapping details. Then `SectionBlocks` can be the child component that represents course sections on the calendar; and `EventBlocks` can be the child component that represents events.

`Modal` is a good example of a parent component and `LoginModal` & `InformationModal` are good examples of child components. Check out how these are implemented!